api-documentation/payments.md

```diff
@@ -46,6 +46,10 @@ Last name of the payee. Required unless invoice\_id is provided.
 Amount in cents that is going to be paid. Required unless invoice\_id is provided. Can't be smaller than `100`.
 {% endapi-method-parameter %}
 
+{% api-method-parameter name="payment\_id" type="string" required=false %}
+Payment identifier, will be used further to reference this Payment. If not provided ClubCollect will generate an identifier (a hexadecimal string of at least 40 chars). This values must be unique and have between 10 and 50 chars so we advise you to provide an UUID like value. If ClubCollect determines that the value is not unique an error will be returned.
```

Comment:
I have just observed that if I provide a payment ID like `5f283622-5e9d-4b43-834f-cb60b2f9426a` (generated using `Ecto.UUID.generate()`) BE accepts it, but it internally converts it silently to `5f283622-5e9d-4b43-834f-cb60b2f9426a` (so it removes the hyphens). Then I'm not able to ask for the status of the payment using the id I have generated.

I think that BE should either not remove the hyphens or return an argument validation error. I would desire the first one as a user.

This should at least be documented, but I think that this is rather a but than a normal behavior.

Reply:
I now see that this is not true. BE seems to not accept `payment_id` and I've got confused with just spotting the last of `-` and not comparing the ids. I think I was to fast starting to test this.

Reply:
@szymon-jez AFIK BE isn't performing any validations on the hyphens. The only requirements to accept the payment ID as valid are the length and the uniqueness. See PR

https://github.com/socialpaymentsbv/billing-engine/pull/9932/files#diff-144ca0bba0d243d9a745c38e0e56d1ea4d102fec7b848f4fa2314662f390785dR173

Reply:
Yes, I was to fast with this. I should have first checked if the BE PR got shipped.